src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts

```diff
@@ -647,9 +641,16 @@ export class XMLHttpRequestController {
   /**
    * Converts this `XMLHttpRequest` instance into a Fetch API `Request` instance.
    */
-  private toFetchApiRequest(): Request {
+  private toFetchApiRequest(
+    body: XMLHttpRequestBodyInit | Document | null | undefined
+  ): Request {
     this.logger.info('converting request to a Fetch API Request...')
 
+    // If the `Document` is used as the body of this XMLHttpRequest,
+    // set its inner text as the Fetch API Request body.
+    const resolvedBody =
```

Comment:
## Bugfix

Sneaking a bugfix here: if the XHR body was a `Document`, its content will now correctly be set on the created request instance.